### ISSUE TITLE
19.03 dmenu / libX11 segfault regression

### ISSUE THREAD

**Issue by hlandau** [NixOS/nixpkgs#59625]
## Issue description
```
$ gdb --args dmenu -i
GNU gdb (GDB) 8.2.1
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-unknown-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from dmenu...(no debugging symbols found)...done.
(gdb) run
Starting program: /nix/store/3l1kjcdhkmi4spcs378cb2nzdpc7g9xl-system-path/bin/dmenu -i                                                                        
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/nix/store/681354n3k44r8z90m35hm8945vsp95h1-glibc-2.27/lib/libthread_db.so.1".                                               
a
b

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7ee0b90 in XmbLookupString () from /nix/store/ycy1gs23m1kdd3rka4fdl5y8h6aasv7h-libX11-1.6.7/lib/libX11.so.6                                        
(gdb) bt
#0  0x00007ffff7ee0b90 in XmbLookupString () from /nix/store/ycy1gs23m1kdd3rka4fdl5y8h6aasv7h-libX11-1.6.7/lib/libX11.so.6                                    
#1  0x0000000000402fab in main ()
(gdb) quit
```

### Steps to reproduce
Run `dmenu -i`, type a few lines (these are menu items) and press Ctrl+D to EOF. The menu appears, as normal. **Type a single character** and a segfault occurs.

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.34, NixOS, 19.03.172138.5c52b25283a (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.03.172138.5c52b25283a"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by livnev** [comment]
@hlandau This is an issue related to input methods, possibly `ibus-anthy` (are you using that?). See [here](http://lists.suckless.org/dev/1902/33294.html) for the upstream bug report discussion.

**Comment by orivej** [comment]
This was a bug introduced in dmenu 4.9. It will be fixed in the next release. The workaround is to unset `XMODIFIERS` for dmenu. Meanwhile I ported the fix from upstream to `master` in 1f16d0496ce86d4c903e2e4ed6cbe4e952a7196f and `release-19.03` in 1f16d0496ce86d4c903e2e4ed6cbe4e952a7196f.
